cg/meta/demultiplex/hk_functions.py

```diff
@@ -0,0 +1,217 @@
+"""Functions interacting with housekeeper in the DemuxPostProcessingAPI."""
+import logging
+from pathlib import Path
+from typing import List, Optional
+from cg.apps.demultiplex.demultiplex_api import DemultiplexingAPI
+from housekeeper.store.models import Version
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.models.demultiplex.flow_cell import FlowCellDirectoryData
+from cg.constants.housekeeper_tags import SequencingFileTag
+from cg.constants.sequencing import Sequencers
+from cg.store import Store
+from cg.meta.demultiplex.utils import (
+    get_lane_from_sample_fastq,
+    get_q30_threshold,
+    get_sample_fastqs_from_flow_cell,
+    get_sample_sheet_path,
+)
+from cg.apps.demultiplex.sample_sheet.read_sample_sheet import (
+    get_sample_internal_ids_from_sample_sheet,
+)
+from cg.utils.files import get_files_matching_pattern
+
+LOG = logging.getLogger(__name__)
+
+
+def store_flow_cell_data_in_housekeeper(
+    flow_cell: FlowCellDirectoryData,
+    hk_api: HousekeeperAPI,
+    demux_api: DemultiplexingAPI,
```

Comment:
Maybe we can consider just sending in the flow cell run directory here? The demux api is not used, only the run dir attribute. I think it is cleaner to just send in that. Might be out of scope for this PR if it was like this already

Reply:
Easier to test if we just send in the run dir as well